### ISSUE TITLE
(Only Firefox Quantum +Windows) Mouse events are broken

### ISSUE THREAD

**Issue by rodri042** [photonstorm/phaser-ce#531]
A bug in the API:
        Phaser version(s): 2.10.3
        Live example: https://codepen.io/anon/pen/QrddjO
        What should happen: It should log mousedown and mouseups on the console, and allow mousemove events while a mouse click is down.
        What happens instead: **ONLY ON WINDOWS + FIREFOX QUANTUM** It never fires mouse down&up events and it doesn't fire *mousemove* events when a button is down.

It's not a Firefox bug because I created another test without Phaser (https://codepen.io/anon/pen/YLNNpy) and it works OK.

**Comment by samme** [comment]
Is it `requestPointerLock()` causing the problem?

**Comment by rodri042** [comment]
> Is it requestPointerLock() causing the problem?

Nope, it happens without calling it too.

> Can you try https://codepen.io/samme/pen/KRaeQa?
> That seems to be working correctly in my Firefox (and just like Chrome). Firefox now supports Pointer Events so it's normal that mousedown and mouseup are not triggered (unless you set game.input.mspointer.capture = false).

Ok, `game.input.mspointer.capture = false` fixed the problem (https://codepen.io/anon/pen/LmxgXM)
I will add that to my codebase, but as a suggestion, I think that `false` should be the default value since it produces these weird behaviors with mouse events.

Or maybe Phaser should log a big warning on input.onDown.add with something like 'HEY, this won't work well unless you disable mspointer capture!!' :stuck_out_tongue:

Thanks for the help!

**Comment by rodri042** [comment]
Yes, but in this example they don't work: https://codepen.io/anon/pen/QrddjO

**Comment by samme** [comment]
In my Firefox, input.onDown and input.onUp are still working after the pointer is locked in <https://codepen.io/anon/pen/QrddjO>.

**Comment by rodri042** [comment]
Are you using Firefox 59.0.2 on Windows? I'm using Windows 7 on a notebook which has a multitouch trackpad, if it helps. I'm always testing with a connected USB mouse.

A user of my game had the same problem with a macbook pro.

**Comment by rodri042** [comment]
> Mine is macOS Firefox 59.0.2.
Oh, yes, the problem is only on Windows. I've tested on Linux and it works OK too.

With pointer lock it doesn't print up/down at all.
Without pointer lock (https://codepen.io/anon/pen/GdWEoJ):
* It print UP/DOWN events, but:
* The movementX and movementY of MOVE events are always the same if a click button is down
* It happens on Chrome too

In this video (https://www.youtube.com/watch?v=7mIJnKdmyS4) I'm holding the left click while moving the mouse, and all the "move" events are fired with the same values.

**Comment by samme** [comment]
> The movementX and movementY of MOVE events are always the same if a click button is down

Check `pointer.isMouse` there. If it's false, then `input.mouse.event` is probably inaccurate. We actually need to add an `event` parameter to that callback.

Can you check <https://codepen.io/samme/pen/KRaeQa> (updated)?

**Comment by rodri042** [comment]
> Can you check https://codepen.io/samme/pen/KRaeQa (updated)?

Same problem: https://www.youtube.com/watch?v=B94jbQcAe0s
While a mouse button is down, movementX and movementY don't change

**Comment by samme** [comment]
What about the last one, under **input.mspointer.event**?

**Comment by rodri042** [comment]
> What about the last one, under input.mspointer.event?

Oh, that one has correct values!

**Comment by samme** [comment]
#265

**Comment by rodri042** [comment]
Good! But I don't undestand something:

How adding the `event` parameter would fix https://codepen.io/anon/pen/QrddjO?

I think that, [in this case](https://www.youtube.com/watch?v=7mIJnKdmyS4), the move callback is never called with correct values while click is down. It should be called with the `movementX` and `movementY` from **input.mspointer.event**, but it's not happening.

**Comment by samme** [comment]
When input.mspointer is running, the new `event` parameter passed to the move callback will be `input.mspointer.event`, which should hold accurate movementX and movementY values.

**Comment by samme** [comment]
> Mouse events can only be prevented when the pointer is down. Hovering pointers (e.g. a mouse with no buttons pressed) cannot have their mouse events prevented. And, the `mouseover` and `mouseout` events are never prevented (even if the pointer is down). 
>
> <https://www.w3.org/Submission/pointer-events/#mapping-for-devices-that-support-hover>

**Comment by samme** [comment]
Maybe Phaser should update the `input.mousePointer.movementX`/`Y` values here.
